File: sample/sample-android/src/main/java/com/example/sampleandroid/view/model/ApiAnnotationScreen.kt
package com.example.sampleandroid.view.model

import androidx.compose.foundation.Text
import androidx.compose.foundation.layout.Column
import androidx.compose.runtime.Composable
import com.example.sampleandroid.view.widget.SampleTextField
import kim.jeonghyeon.androidlibrary.compose.Screen
import kim.jeonghyeon.androidlibrary.compose.unaryPlus
import kim.jeonghyeon.androidlibrary.compose.widget.Button
import kim.jeonghyeon.androidlibrary.extension.resourceToString
import kim.jeonghyeon.sample.compose.R
import kim.jeonghyeon.sample.viewmodel.ApiAnnotationViewModel

@Composable
fun ApiAnnotationScreen(model: ApiAnnotationViewModel) {
    Screen(model) {
        Column {
            SampleTextField("Input value", model.input)
            Button("update") { model.onClick() }
            Text("current value : ${+model.result}")
        }
    }
}

File: sample/sample-native/kotlinIOS/screen/model/ApiAnnotationScreen.swift
//
//  ApiAnnotationScreen.swift
//  kotlinIOS
//
//  Created by hyun kim on 08/08/20.
//  Copyright © 2020 hyun kim. All rights reserved.
//

import Foundation
import SwiftUI
import sample_base


func ApiAnnotationScreen(_ model: ApiAnnotationViewModel) -> some View {
    Screen(model) {
        Column(alignment: .center) {
            Text("current value : \(+model.result)")
            SampleTextField("update", model.input).frame(width: 100, alignment: .center)
            Button("Update") {
                model.onClick()
            }
        }
    }
}
